test cases/common/145 special characters/meson.build

```diff
@@ -42,29 +42,47 @@ gen2 = custom_target('gen2',
 # characters in filenames or target names)
 #
 # TODO: similar tests needed for languages other than C
+# TODO: add similar test for quote and doublequote, carefully
 
-special = [['dollar', '$'],
-           ['hash', '#'],
-           ['quote', '\''],
-           ['doublequote', '"'],
-           ['lt', '<'],
-           ['gt', '>'],
-           ['amp', '&'],
-           ['slash', '/'],
-           ['backslash', '\\']]
+special = [
+  ['amp', '&'],
+  ['at', '@'],
+  ['backslash', '\\'],
+  ['dollar', '$'],
+  ['gt', '>'],
+  ['hash', '#'],
+  ['lt', '<'],
+  ['slash', '/'],
+]
+
+# set to true if you want the builder to go green, false to see the bugs in action
+skipbugs = true
+cc = meson.get_compiler('c')
 
 foreach s : special
-  args = '-DCHAR="@0@"'.format(s[1])
-  e = executable('arg-string-' + s[0], 'arg-string-test.c', c_args: args)
-  test('arg-string-' + s[0], e, args: s[1])
+  if skipbugs and (s[0] == 'hash') and (cc.get_id() == 'msvc')
+    message('passing the hash char fails always at runtime with msvc (CHAR is "="?!, arg is ok)')
```

Comment:
Fascinating.  I don't recall tripping over that before.

According to https://docs.microsoft.com/en-us/cpp/build/reference/d-preprocessor-definitions?view=vs-2019, `'#` is permitted instead of `=` in `/D`, which is probably related somehow.

Reply:
Yep.  I removed the test for # and left a comment behind.